orttraining/orttraining/core/session/training_session.h

```diff
@@ -32,9 +32,15 @@ class TrainingSession : public InferenceSession {
    * Partition information of each paritioned weight
    */
   struct PartitionInfo {
+    // value of the original shape of the weight
```

Comment:
One thing we should think earlier，probabbly not now, for a bit later is: we will make change to the naming or this schema of weight partition information, once it goes to more stable, we need some flags to control old logic parsing/aggregation ckpts I guess. What we can do currently is, to be aware of this possibility, and try to make it stable as early as possible.

Reply:
that makes sense. Having said that, the current fields saved as a part of the checkpoints will most likely not change/be dropped. Even if we add more fields to the cpp struct, what we are essentially returning to the frontend are the original shape and the megatron row partition fields. It is possible that there might be additional fields we would need to save(like the 'stride' for megatron), and we need to make sure we are backward compatible for the frontend aggregation code.